internal/ctlog/ctlog.go

```diff
@@ -376,6 +387,7 @@ type UploadOptions struct {
 
 var optsHashTile = &UploadOptions{Immutable: true}
 var optsDataTile = &UploadOptions{Compress: true, Immutable: true}
+var optsStaging = &UploadOptions{Compress: true, Immutable: true}
```

Comment:
Oh that's a good observation. One solution is to use the hash of the whole checkpoint that includes a timestamp, but I actually think we can show that this is fine regardless: we are worried about two staging bundles for N->M and N+k->M conflicting; the only case in which this can happen is if both N and N+k are successfully fully sequenced, because we never make progress without a full sequencing. The good news is that I think both the tiles in N->M and the tiles in N+k->M are valid to make a M tree whole from N+k, so it doesn't matter which one "wins".

Reply:
I tried writing a test for this in 4df346dbbaf3fa8783e9d03ed614ce1a40925d5d but I actually think it's impossible: even if N -> N+2 and N -> N+1 manage to have the same timestamp (which is part of the leaf and so part of the tree hash and so part of the staging path) then N+1 -> N+2 can't have the same timestamp because time is required to move forward at each sequencing.

Reply:
I had to mark staging uploads not immutable though, because there are differences in the serialized values that I have actually not yet figured out.